core/agents/developer.py

```diff
@@ -287,8 +281,8 @@ async def ask_to_execute_task(self) -> bool:
         if user_response.cancelled or user_response.button == "skip":
             log.info(f"Skipping task: {description}")
             self.next_state.current_task["instructions"] = "(skipped on user request)"
-            self.next_state.complete_task()
-            await self.send_message(f"Skipping task {description}")
+            self.next_state.current_task["status"] = "skipped"
```

Comment:
This is a modification of a structure (dictionary within a list) inside the SQLALchemy ORM field. SQLalchemy won't detec it so it's very likely it won't make an update in the database.

For this reason, we have [`ProjectState.flag_tasks_as_modified()`](https://github.com/Pythagora-io/gpt-pilot/blob/49cb71121c6944934d04ea9a904ead974c287389/core/db/models/project_state.py#L282), to tell SQLalchemy the field should be updated.

And since the task status is being set in several places, I'd suggest creating `set_current_task_status()` helper method in ProjectState which would set it and also call flag modified so nobody forgets it somewhere.